R/bayesfactor_inclusion.R

```diff
@@ -0,0 +1,199 @@
+#' Get inclusion BFs for effects across Bayesian models
+#'
+#'
+#' @author Mattan S. Ben-Shachar
+#' @param models an object of class \code{BFGrid} or \code{BFBayesFactor}.
+#' @param match_models If \code{FALSE} (default), Inclustion BFs are computed by
+#' comparing all models with an effect against all models without the effect. If \code{TRUE},
+#' Inclusion BFs are computed by comparing all models with an effect against models without
+#' the effect AND without any higher-order interactions with the effect.
+#' @param prior_odds optional vector of prior odds for the models. See \code{BayesFactor::priorOdds}
+#'
+#' @return a data frame containing the prior and posterior probabilities, and BF (log) for each effect.
+#'
+#' @note Random effects in the \code{lme} style will be displayed as interactions:
+#' i.e., \code{(X|G)} will become \code{1:G} and \code{X:G}.
+#'
+#' @examples
+#' \dontrun{
+#' library(bayestestR)
+#' library(BayesFactor)
+#'
+#' BF <- generalTestBF(len ~ supp * dose, ToothGrowth, progress = FALSE)
+#'
+#' bayesfactor_inclusion(BF)
+#'
+#' # compare only matched models:
+#' bayesfactor_inclusion(BF, match_models = TRUE)
+#' }
+#'
+#' @references Hinne, M., Gronau, Q. F., van den Bergh, D., & Wagenmakers,
+#' E. (2019, March 25). A conceptual introduction to Bayesian Model
+#' Averaging. https://doi.org/10.31234/osf.io/wgb64
+#'
+#'#' Clyde, M. A., Ghosh, J., & Littman, M. L. (2011). Bayesian
+#' adaptive sampling for variable selection and model averaging. Journal
+#' of Computational and Graphical Statistics, 20(1), 80-101.
+#'
+#' Mathot. S. (2017). Bayes like a Baws:
+#' Interpreting Bayesian Repeated Measures in JASP [Blog post].
+#' Retrieved from https://www.cogsci.nl/blog/interpreting-bayesian-repeated-measures-in-jasp
+#'
+#'
+#' @export
+bayesfactor_inclusion <- function(models, match_models = FALSE, prior_odds = NULL) {
+  UseMethod("bayesfactor_inclusion")
+}
+
+bayesfactor_inclusion.BFGrid <- function(models, match_models = FALSE, prior_odds = NULL) {
+  # Build Models Table #
+  df.model <- get_model_table(models, priorOdds = prior_odds)
+  effnames <- colnames(df.model)[-(1:3)]
+
+  # Build Interaction Matrix #
+  if (match_models) {
+    df.interaction <- data.frame(effnames)
+
+    for (eff in effnames) {
+      df.interaction[,eff] <- sapply(effnames, function(x) includes_interaction(x,eff))
+    }
+    rownames(df.interaction) <- effnames
+    df.interaction <- df.interaction[,-1]
+  }
+
+  # Build Effect Table #
+  df.effect <- data.frame(effnames,
```

Comment:
effnames might be a factor, unless you set `stringsAsFactors = FALSE` here. Could you check if this is working as intended?

Reply:
Why oh why is `stringsAsFactors = TRUE` by default?

Reply:
It works any way, but I set `stringsAsFactors = FALSE` just to be safe.